File: src/main/java/no/novari/flyt/history/validation/OnlyOneStatusFilter.java
package no.novari.flyt.history.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = OnlyOneStatusFilterValidator.class)
public @interface OnlyOneStatusFilter {

    String message() default "contains both latest status events and status";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/history/validation/OnlyOneStatusFilter.kt
package no.novari.flyt.history.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [OnlyOneStatusFilterValidator::class])
annotation class OnlyOneStatusFilter(
    val message: String = "contains both latest status events and status",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
